packages/connect-node/src/node-universal-header.ts

```diff
@@ -78,10 +78,19 @@ export function webHeaderToNodeHeaders(
     }
   } else if ("forEach" in headersInit) {
     if (typeof headersInit.forEach == "function") {
-      headersInit.forEach((value, key) => {
-        const k = key.toLowerCase();
-        o[k] = value;
-      });
+        const setCookieHeaders: string[] = [];
+        headersInit.forEach((value, key) => {
+            const k = key.toLowerCase();
+            if (k === 'set-cookie') {
+                setCookieHeaders.push(value);
+            }
+            else {
+                o[k] = value;
+            }
+        });
+        if (setCookieHeaders.length > 0) {
+            o['set-cookie'] = setCookieHeaders;
+        }
```

Comment:
This works, but I believe we have to handle it for the array (line 74) and the object literal (line 95) as well.

What do you think about adding a local function that adds multiple entries for the same key as an array?

```ts
  const append = (key: string, value: string): void => {
    key = key.toLowerCase();
    const existing = o[key];
    if (typeof existing == "string") {
      o[key] = [existing, value];
    } else if (Array.isArray(existing)) {
      existing.push(value);
    } else {
      o[key] = value;
    }
  };
```

In the forEach callback, you would simply call `append(key, value)`, and you could do the same for array and object literal.

Reply:
Yes, that would make sense, but I believe those other two cases are intended to handle the header objects from the http and http2 packages? In both of those, you have option to set cookie value that is an array of strings, which will give the desired effect for set-cookie here. In particular, it's from the fetch API Headers object (which is used for headers in connect-express), where we don't have a way to create an array value where we need this special handling to create the array for us. The fetch API even introduced a special getSetCookie method for this purpose, which we could also use here instead, but it would cause a typing error: https://github.com/microsoft/TypeScript/issues/55270

Very happy with whatever way you'd prefer to structure this, but main thing that would fill the gap for us is something that handles multiple set-cookie headers when using fetch api.

Reply:
Great point, addressed with last commit.